### PR TITLE
Add tests for performance of data collectors

### LINKED ISSUE #2670: Add test functions for collectors to test performance
Having a few test functions that can test the `gen_data_collector` , `gen_kw_collector` etc will save us some time when working on opptimizing them.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #2670 

## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: tests/libres_tests/res/enkf/export/test_performance.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ 
+    def test_gen_data_collector_snake_oil(self):
+        with ErtTestContext(
+            "python/enkf/export/gen_data_collector", self.snake_oil_config
+        ) as context:
+            ert = context.getErt()
+
+            df: DataFrame= self.benchmark(GenDataCollector.loadGenData, ert, "default_0", "SNAKE_OIL_OPR_DIFF", 199)
+  
+            # start = time.perf_counter()
```
Comment (oysteoh) [inline tests/libres_tests/res/enkf/export/test_performance.py:57]: is the intention to uncomment these at some point ?

Comment (frode-aarstad) [reply tests/libres_tests/res/enkf/export/test_performance.py]: Removed the comments

File: tests/libres_tests/res/enkf/export/test_performance.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import pytest
+import time
```
Comment (oysteoh) [inline tests/libres_tests/res/enkf/export/test_performance.py:2]: `time` is not used anywhere i think?

**Comment by oysteoh** [comment]
Jenkins: Test this please!

File: tests/libres_tests/res/enkf/export/test_performance.py (hunk 2 above)
Comment (frode-aarstad) [reply tests/libres_tests/res/enkf/export/test_performance.py:2]: I will remove it

**Comment by frode-aarstad** [comment]
Jenkins test this

**Comment by frode-aarstad** [comment]
Jenkins test this please

**Review by oysteoh — APPROVED** [review]

**Pull request merged**